### ISSUE TITLE
Cannot change asset type from 'fiat' to 'ethereum token'

### ISSUE THREAD

**Issue by rkalis** [rotki/rotki#3479]
## Problem Definition

I accidentally added PUDGY (https://www.coingecko.com/en/coins/pudgy-vault-nftx) as a 'fiat' asset. When I tried to change it to an 'ethereum token' instead, I got the message 'Tried to edit non existing ethereum token with address 0xabea7663c472648d674bd3403d94c858dfeef728'. I ended up deleting the asset and re-adding it, but it seems like it should be possible to edit the asset type. Screenshot is attached.

### System Description

Operating system: macOS 11.4
Rotki version: 1.20.1

<img width="1190" alt="Screen Shot 2021-09-18 at 17 02 48" src="https://user-images.githubusercontent.com/9142187/133893369-a1defc3e-1ef7-4357-b4b4-7d025247cb7d.png">

**Comment by LefterisJP** [comment]
We can probably give a better error message. But yes you can't edit the type of an asset. This is how the system is designed. On the type depend all the fields of the asset and its unique identifier. This is something that can't be edited.

**Comment by rkalis** [comment]
I see. Maybe it should be greyed out then similar to how the "label" is greyed out when editing a manual balance?

**Comment by LefterisJP** [comment]
> I see. Maybe it should be greyed out then similar to how the "label" is greyed out when editing a manual balance?

Or perhaps just not allow ethereum token in there. IT's in there? I would have to check and see. If yes then yeah we need it out.

**Comment by rkalis** [comment]
Yeah the reason I thought it *should* be possible is that the UI presented it as possible. It only errored out when I tried to click "save" after filling the details (like address).

**Comment by LefterisJP** [comment]
> Yeah the reason I thought it _should_ be possible is that the UI presented it as possible. It only errored out when I tried to click "save" after filling the details (like address).

Okay reopened the issue. We should address this.

**Comment by kelsos** [comment]
@LefterisJP So the issue here is that you should not be able to change the token type after adding a token, but the frontend allows you to do so?

**Comment by LefterisJP** [comment]
the type of a token should not be changeable. As the unique id is set then. The frontend allowing you to do so is one thing, but wanted to also check what does the backend do. And what request is done and what is returned in this case.

**Comment by kelsos** [comment]
Ok, the frontend part should be really easy to fix, we just need to disable the dropdown on edit.

**Comment by LefterisJP** [comment]
Putting this in the next patch release. Should be easy to handle.

**Comment by kelsos** [comment]
Do we also need backend changes here or is it only the frontend?_

**Comment by LefterisJP** [comment]
frontend alone should fix it. But will have to do a backend check to see the error the user got. Something sounded strange.
